pkg/kubelet/kubelet.go

```diff
@@ -2555,6 +2564,136 @@ func (kl *Kubelet) fastStatusUpdateOnce() {
 	}, 100*time.Millisecond, stopCh)
 }
 
+// handleMaxContainerCheckpoints goes through the list of checkpoints of a
+// container in a pod and deletes old checkpoints if there are more than
+// 'kl.maxContainerCheckpoints'. To track the number of checkpoints a JSON file
+// is created and updated for each new checkpoint. If a new checkpoint would
+// result in more than 'kl.maxContainerCheckpoints' the oldest checkpoint will
+// be removed.
+func (kl *Kubelet) handleMaxContainerCheckpoints(
+	podUID types.UID,
+	podFullName,
+	containerName string,
+	options *runtimeapi.CheckpointContainerRequest,
+	checkpointTime int64,
+) error {
+
+	if kl.maxContainerCheckpoints <= 0 {
+		// This cannot really happen as the kubelet config parser will
+		// set this to 10 as a default value if 0 or not set. The error
+		// handling is mainly here to make sure tests are working.
+		return fmt.Errorf("maxContainerCheckpoints needs to > 0")
+	}
+
+	type checkpoint struct {
+		Archive   string `json:"archive,omitempty"`
+		Size      int64  `json:"size,omitempty"`
+		Timestamp int64  `json:"timestamp,omitempty"`
+	}
+	type containerCheckpointMetadata struct {
+		PodFullName   string       `json:"podFullName,omitempty"`
+		ContainerName string       `json:"containerName,omitempty"`
+		TotalSize     int64        `json:"totalSize,omitempty"`
+		Checkpoints   []checkpoint `json:"checkpoints"`
+	}
```

Comment:
Are those type definitions part of any library we could use? If not then we can make them public to be usable from the tests directly.